### DISCUSSION TITLE
Lazy routes vs non-lazy routes

### DISCUSSION THREAD

**Discussion by gschwa** [TanStack/router#1178] [General]
We would like to utilize lazy routes to reduce the size of the initial bundle but `createLazyFileRoute` has significantly less options than `createFileRoute`. Specifically, the hooks like `onEnter`, validating search params, etc. How do you get the best of both worlds?

**Comment by Semigradsky** [comment]
https://tanstack.com/router/latest/docs/framework/react/guide/code-splitting#after

It looks like you can have two config in one time: posts.tsx and posts.lazy.tsx e.g.

**Reply by gschwa** [reply]
Thank you! It was not immediately obvious that you would define it both ways. Maybe in the future the API can be simpler.
